module/solutions/helper.js

```diff
@@ -142,4 +1012,2537 @@ module.exports = class SolutionsHelper {
+                {
+                    "_id":"62e228eedd8c6d0009da5084",
+                    "solutionId":"627dfc6509446e00072ccf78",
+                    "surveyId":"62e228eedd8c6d0009da507d",
+                    "status":"completed",
+                    "surveyInformation":{
+                        "name":"Create a Survey (To check collated reports) for 4.9 regression -- FD 380",
+                        "description":"Create a Survey (To check collated reports) for 4.9 regression -- FD 380"
+                    }
+                }
+            ]
+          }       
+           */
+          // let surveySubmissionDetails =
+          //   await surveyService.userSurveySubmissions(
+          //     userToken,
+          //     solutionData.solutionId
+          //   );
+          // let surveySubmissionData = surveySubmissionDetails.result;
+          // if (surveySubmissionData.length > 0) {
+          //   checkForTargetedSolution.result.submissionId =
+          //     surveySubmissionData[0]._id ? surveySubmissionData[0]._id : "";
+          //   checkForTargetedSolution.result.surveyId = surveySubmissionData[0]
+          //     .surveyId
+          //     ? surveySubmissionData[0].surveyId
+          //     : "";
+          //   checkForTargetedSolution.result.submissionStatus =
+          //     surveySubmissionData[0].status;
+          // } else if (!isSolutionActive) {
+          //   throw new Error(CONSTANTS.apiResponses.LINK_IS_EXPIRED);
+          // }
+        } else if (solutionData.type === CONSTANTS.common.IMPROVEMENT_PROJECT) {
+          // Targeted solution
+          if (
+            checkForTargetedSolution.result.isATargetedSolution &&
+            createProject
+          ) {
+            //targeted user with project creation
+
+            let projectDetailFromLink =
+              await projectQueries.projectDocument(
+                {"solutionId" : solutionId}
+              );
+
+            if (!projectDetailFromLink || !projectDetailFromLink.data) {
+              return resolve(projectDetailFromLink);
+            }
+            if (projectDetailFromLink.data.length < 1 && !isSolutionActive) {
+              throw new Error(CONSTANTS.apiResponses.LINK_IS_EXPIRED);
+            }
+
+            checkForTargetedSolution.result["projectId"] = projectDetailFromLink
+              .data._id
+              ? projectDetailFromLink.data._id
+              : "";
+          } else if (
+            checkForTargetedSolution.result.isATargetedSolution &&
+            !createProject
+          ) {
+            //targeted user with no project creation
+            let findQuery = {
+              userId: userId,
+              projectTemplateId: solutionData.projectTemplateId,
+              referenceFrom: {
+                $ne: CONSTANTS.common.LINK,
+              },
+              isDeleted: false,
+            };
+
+            let checkTargetedProjectExist =
+              await projectQueries.projectDocument(
```

Comment:
@Prajwal17Tunerlabs remove user token and make the changes in the next line as you are not getting the success and data and all